Add NAPPN 2021 workshop materials (#246)
terraref/tutorials: videos/nappn_2021/walkthrough.Rmd

---
title: "NAPPN 2021 TERRA REF workshop notes"
author: "Kristina Riemer"
output: github_document
urlcolor: blue
---

# Workshop details

Title: Using TERRA REF high throughput, sensor-collected plant data with R

Date: Tuesday February 16, 2021

Time: 8:30am - 11:00am CST

## Agenda

- 7:30 - 8:00: Installation time/troubleshooting

> - 7:30 - 8:00: Overview of TERRA REF experiment

Author: I figured we could spent the first couple minutes pointing folks towards the installation instructions if they haven't done so already, and then jump right into the data intro? And I put the times in AZ time for our sake, I'll add them in EST too.

    - R packages: raster, dplyr, ggplot2
4. Download Dryad data
    - Download manually at [https://datadryad.org/stash/dataset/doi:10.5061/dryad.4b8gtht99](https://datadryad.org/stash/dataset/doi:10.5061/dryad.4b8gtht99) by clicking blue "Download dataset" button OR
    - Download with R using code below

```{r, eval=FALSE}
local_path <- "dryad_data.zip"
download.file(url = "https://datadryad.org/stash/downloads/file_stream/624637", destfile = local_path)
unzip(local_path)
```

## Data overview

The purpose of this walkthrough is to provide an introduction the TERRA REF public domain high throughput phenomics datasets (LeBauer et al 2020), and to give learners a chance to access and explore a representative cross section of these data. 

In addition to this workshop, we have online tutorials for using these data at [https://terraref.github.io/tutorials/](https://terraref.github.io/tutorials/). There are also materials from previous walkthrough workshops at [https://github.com/terraref/tutorials/tree/master/videos](https://github.com/terraref/tutorials/tree/master/videos). We are striving to make these data as accessible and usable as possible. 

> TERRA REF is a project funded by the Advanced Research Projects Agency for Energy (ARPA-E). The objective is to advance the science and technology of high throughput field phenotyping, which is the automated measurement of plants. TERRA REF has deployed an exceptionally large robot capable of collecting data from a range of sensors at 1 mm spatial resolution every other day. You can learn more about the project from the website, terraref.org. The suite of sensors include visible light cameras, hyperspectral cameras, a laser 3d scanner, and environmental data. In addition there are weather stations and high resolution genome sequences for hundreds of varieties of Sorghum being monitored. 

Reviewer: ```suggestion
TERRA REF is a project funded by the Advanced Research Projects Agency for Energy (ARPA-E). The objective is to advance the science and technology of high throughput field phenotyping, which is the automated measurement of plant traits. The TERRA REF field scanner is the world’s largest agricultural robot and is capable of collecting data from a range of sensors at 1 mm spatial resolution. You can learn more about the project from the website, [terraref.org](https://terraref.org) and the documentation, [docs.terraref.org](https://docs.terraref.org). The suite of sensors include visible light cameras, hyperspectral cameras, a laser 3d scanner, and environmental data. These are provided alongside a suite of traits collected by hand as well as derived from sensors. Along with these data are high resolution genome sequences for hundreds of varieties of Sorghum being monitored as well as soil, weather, and other environmental data.
```

### Experimental design

> (Show picture of site) Field site is in Arizona. Field is passed over by large robot called a gantry. Lot of equipment in hanging box to collect many types of data. 

Reviewer: provide both a view of the gantry and a screenshot from the traitvis webapp with plots outlined ... I will have these in my slides and can share them.

> Gantry goes systematically over entire field once a day. Some sensors take data every day, like 9,000 images from RGB camera. Others more intermittently, like hyperspectral images, because of data space limits. 

Reviewer: ```suggestion
The field scanner systematically scans the entire field, typically every other day during the growing season, but in some cases multiple times per day.
```

This is a good place for the figure of sensor data products x time from the README

Author: I present that figure during the **Dryad data** section later on, when introducing the README, but should it go here instead/also?

Reviewer: should be fine down there

Sensors include: 

> * camera that takes pairs of red-green-blue images (**Stereo RGB**)

Reviewer: ```suggestion
* camera that takes stereo pairs of red-green-blue images (**Stereo RGB**)
```

* thermal infrared images (**FLIR**)

> * images at a bunch of wavelengths to get hyperspectral data (**VNIR/SWIR**)

Reviewer: ```suggestion
* hyperspectral images of over 1000 wavelengths at 2/3 (VNIR) or 12 (SWIR) nm spectral resolution (**VNIR/SWIR**)
```

> * laser that collects points on plant surfaces to create 3D image (**3D Laser**)

Reviewer: ```suggestion
* laser that produces a depth map image that is converted to 3D point clouds (**3D Laser**)
```

> * measures plant fluorescence (**PS II Fluor**)

Reviewer: ```suggestion
* time series of 100 images capturing plant fluorescence response (**PS II Fluor**)
```

> * handful of others, including environmental data such as temperature and light

Reviewer: ```suggestion
* other data, including environmental data such as temperature and light
```

Collecting data since 2015, and are up to 8 seasons worth in that time. Originally for sorghum, but now open to other crop species and organizations that want to use system.

> 

Reviewer: Provide core details of season 4 and 6 experiments ... that these are currently in the public domain,  are Sorghum & etc, that they are trials of the biomass assoc. panel, planting and harvest dates, # of plots. Can lean on README and documentation for this information.

### Traitvis webapp

> The easiest way to start to visualize and understand the TERRA REF data is through our webapp. Go to [https://traitvis.workbench.terraref.org/](https://traitvis.workbench.terraref.org/), which takes a minute to load. Displays plots of various data across collection time.

Reviewer: ```suggestion
The easiest way to start to visualize and understand the TERRA REF data is through the Traitvis shiny app website. Go to [terraref.org/traitvis/](https://terraref.org/traitvis/); this takes a minute to load because there is so much data, and _may freeze if everyone does it at once_! It displays time plots of trait data as well as maps and thumbnails of field scans.
```

Author: I think whether or not we encourage attendees to actually go to the traitvis webapp themselves depends on how many people there are? If there are a lot of people, we should recommend they just watch and go to the site some time later.

> As example, we’ll look at data from Season 6, which we looked at last week, by going to “MAC Season 6” tab.

Reviewer: Put screenshots here because app will likely crash in a course such as this

- 10:35 - 11:00: Using sensor data from Globus
- 11:00 - 11:25: Using trait data with R
- 11:25 - 11:30: break
- 11:30 - 11:50: Using weather data with R, combining trait & weather data
- 11:50 - 12:00: TERRA REF data review, wrap-up

AZ time: 

- 7:30 - 8:00: Installation time/troubleshooting
- 7:30 - 8:00: Overview of TERRA REF experiment
- 8:00 - 8:30: Downloading sensor data from Globus
- 8:30 - 8:35: break
- 8:35 - 9:00: Using sensor data from Globus
- 9:00 - 9:25: Using trait data with R
- 9:25 - 9:30: break
- 9:30 - 9:50: Using weather data with R, combining trait & weather data
- 9:50 - 10:00: TERRA REF data review, wrap-up

### Download/install - Kristina

1. Sign up for Globus account on [https://www.globus.org/](https://www.globus.org/)
2. Download Globus Connect Personal
    - At [https://www.globus.org/globus-connect-personal](https://www.globus.org/globus-connect-personal)
    - Follow automated setup steps
3. Download R and RStudio
    - R at [https://www.r-project.org/](https://www.r-project.org/)
    - RStudio at [https://rstudio.com/products/rstudio/download/](https://rstudio.com/products/rstudio/download/)
    - R packages: raster, dplyr, ggplot2

> 4. Download Dryad data

Reviewer: ```suggestion
4. Download data from Dryad:
    - full package is 723 MB; this could take an hour on a slow (2Mbps) internet connection
```

>     - Download manually at [https://datadryad.org/stash/dataset/doi:10.5061/dryad.4b8gtht99](https://datadryad.org/stash/dataset/doi:10.5061/dryad.4b8gtht99) by clicking blue "Download dataset" button OR

Reviewer: ```suggestion
    - Download manually at [https://datadryad.org/api/v2/datasets/doi%3A10.5061%2Fdryad.4b8gtht99/download](https://datadryad.org/api/v2/datasets/doi%3A10.5061%2Fdryad.4b8gtht99/download) OR
```

Author: What is the difference in these links?

    - Download with R using code below

```{r, eval=FALSE}
local_path <- "dryad_data.zip"

> dryad_url <- ""

Reviewer: ```suggestion
dryad_url <- "https://datadryad.org/api/v2/files/624637/download"
```
